File: rnapolis-frontend/src/app/entity/publication.ts
export class Publication {
  id: string;
  authors: string;
  title: string;
  journal: string;
  volumeIssue: string;
  year: number;
  pages: string;


  constructor(authors: string, title: string, journal: string, volumeIssue: string, year: number, pages: string) {
    this.authors = authors;
    this.title = title;
    this.journal = journal;
    this.volumeIssue = volumeIssue;
    this.year = year;
    this.pages = pages;
  }
}

File: rnapolis-backend/src/main/java/rnapolis/models/Publication.java
package rnapolis.models;

import java.time.Instant;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "publications")
@Data
@Builder
public class Publication {

    @Id
    private String id;

    private String authors;
    private String title;
    private String editors;
    private String journal;
    private String volumeIssue;
    private String publishers;
    private Integer year;
    private String pages;
    private String link;

    @CreatedDate
    private Instant createdDate;

    @LastModifiedDate
    private Instant lastModifiedDate;
}
